From: Jeff Mahoney <jeffm@suse.com>
Subject: kabi: powerpc: move other fields to paca_aux
Patch-mainline: No, kabi
References: bsc#1114763

patches.suse/powerpc-rfi-flush-Move-RFI-flush-fields-out-of-the-p.patch moved
the RFI-related fields to paca_aux_struct.  This patch moves the rest.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>
---
 arch/powerpc/include/asm/paca.h        |   18 +++++++++---------
 arch/powerpc/mm/slb.c                  |    6 +++---
 arch/powerpc/platforms/pseries/ras.c   |   14 +++++++-------
 arch/powerpc/platforms/pseries/setup.c |    4 ++--
 4 files changed, 21 insertions(+), 21 deletions(-)

--- a/arch/powerpc/include/asm/paca.h
+++ b/arch/powerpc/include/asm/paca.h
@@ -56,6 +56,15 @@ struct paca_aux_struct {
 	u64 exrfi[13] __aligned(0x80);
 	void *rfi_flush_fallback_area;
 	u64 l1d_flush_size;
+#ifdef CONFIG_PPC_PSERIES
+	u8 *mce_data_buf;		/* buffer to hold per cpu rtas errlog */
+#endif /* CONFIG_PPC_PSERIES */
+
+#ifdef CONFIG_PPC_BOOK3S_64
+	/* Capture SLB related old contents in MCE handler. */
+	struct slb_entry *mce_faulty_slbs;
+	u16 slb_save_cache_ptr;
+#endif /* CONFIG_PPC_BOOK3S_64 */
 };
 #endif
 
@@ -217,15 +226,6 @@ struct paca_struct {
 #endif
 	struct kvmppc_host_state kvm_hstate;
 #endif
-#ifdef CONFIG_PPC_PSERIES
-	u8 *mce_data_buf;		/* buffer to hold per cpu rtas errlog */
-#endif /* CONFIG_PPC_PSERIES */
-
-#ifdef CONFIG_PPC_BOOK3S_64
-	/* Capture SLB related old contents in MCE handler. */
-	struct slb_entry *mce_faulty_slbs;
-	u16 slb_save_cache_ptr;
-#endif /* CONFIG_PPC_BOOK3S_64 */
 };
 
 extern void copy_mm_to_paca(struct mm_struct *mm);
--- a/arch/powerpc/mm/slb.c
+++ b/arch/powerpc/mm/slb.c
@@ -196,7 +196,7 @@ void slb_save_contents(struct slb_entry
 	unsigned long e, v;
 
 	/* Save slb_cache_ptr value. */
-	get_paca()->slb_save_cache_ptr = get_paca()->slb_cache_ptr;
+	get_paca()->aux_ptr->slb_save_cache_ptr = get_paca()->slb_cache_ptr;
 
 	if (!slb_ptr)
 		return;
@@ -250,9 +250,9 @@ void slb_dump_contents(struct slb_entry
 	pr_err("----------------------------------\n");
 
 	/* Dump slb cache entires as well. */
-	pr_err("SLB cache ptr value = %d\n", get_paca()->slb_save_cache_ptr);
+	pr_err("SLB cache ptr value = %d\n", get_paca()->aux_ptr->slb_save_cache_ptr);
 	pr_err("Valid SLB cache entries:\n");
-	n = min_t(int, get_paca()->slb_save_cache_ptr, SLB_CACHE_ENTRIES);
+	n = min_t(int, get_paca()->aux_ptr->slb_save_cache_ptr, SLB_CACHE_ENTRIES);
 	for (i = 0; i < n; i++)
 		pr_err("%02d EA[0-35]=%9x\n", i, get_paca()->slb_cache[i]);
 	pr_err("Rest of SLB cache entries:\n");
--- a/arch/powerpc/platforms/pseries/ras.c
+++ b/arch/powerpc/platforms/pseries/ras.c
@@ -371,7 +371,7 @@ static irqreturn_t ras_error_interrupt(i
 
 static inline struct rtas_error_log *fwnmi_get_errlog(void)
 {
-	return (struct rtas_error_log *)local_paca->mce_data_buf;
+	return (struct rtas_error_log *)local_paca->aux_ptr->mce_data_buf;
 }
 
 /*
@@ -407,18 +407,18 @@ static struct rtas_error_log *fwnmi_get_
 
 	h = (struct rtas_error_log *)&savep[1];
 	/* Use the per cpu buffer from paca to store rtas error log */
-	memset(local_paca->mce_data_buf, 0, RTAS_ERROR_LOG_MAX);
+	memset(local_paca->aux_ptr->mce_data_buf, 0, RTAS_ERROR_LOG_MAX);
 	if (!rtas_error_extended(h)) {
-		memcpy(local_paca->mce_data_buf, h, sizeof(__u64));
+		memcpy(local_paca->aux_ptr->mce_data_buf, h, sizeof(__u64));
 	} else {
 		int len, error_log_length;
 
 		error_log_length = 8 + rtas_error_extended_log_length(h);
 		len = min_t(int, error_log_length, RTAS_ERROR_LOG_MAX);
-		memcpy(local_paca->mce_data_buf, h, len);
+		memcpy(local_paca->aux_ptr->mce_data_buf, h, len);
 	}
 
-	return (struct rtas_error_log *)local_paca->mce_data_buf;
+	return (struct rtas_error_log *)local_paca->aux_ptr->mce_data_buf;
 }
 
 /* Call this when done with the data returned by FWNMI_get_errinfo.
@@ -536,7 +536,7 @@ static void pseries_print_mce_info(struc
 #ifdef CONFIG_PPC_BOOK3S_64
 	/* Display faulty slb contents for SLB errors. */
 	if (error_type == MC_ERROR_TYPE_SLB)
-		slb_dump_contents(local_paca->mce_faulty_slbs);
+		slb_dump_contents(local_paca->aux_ptr->mce_faulty_slbs);
 #endif
 
 	printk("%s%s Machine check interrupt [%s]\n", level, sevstr,
@@ -611,7 +611,7 @@ static int mce_handle_error(struct rtas_
 			 * level of recursion.
 			 */
 			if (local_paca->in_mce == 1)
-				slb_save_contents(local_paca->mce_faulty_slbs);
+				slb_save_contents(local_paca->aux_ptr->mce_faulty_slbs);
 			flush_and_reload_slb();
 			disposition = RTAS_DISP_FULLY_RECOVERED;
 			rtas_set_disposition_recovered(errp);
--- a/arch/powerpc/platforms/pseries/setup.c
+++ b/arch/powerpc/platforms/pseries/setup.c
@@ -131,7 +131,7 @@ static void __init fwnmi_init(void)
 	mce_data_buf = __va(memblock_alloc_base(RTAS_ERROR_LOG_MAX * nr_cpus,
 					RTAS_ERROR_LOG_MAX, ppc64_rma_size));
 	for_each_possible_cpu(i) {
-		paca[i].mce_data_buf = mce_data_buf +
+		paca[i].aux_ptr->mce_data_buf = mce_data_buf +
 						(RTAS_ERROR_LOG_MAX * i);
 	}
 
@@ -141,7 +141,7 @@ static void __init fwnmi_init(void)
 	slb_ptr = __va(memblock_alloc_base(size, sizeof(struct slb_entry),
 					   ppc64_rma_size));
 	for_each_possible_cpu(i)
-		paca[i].mce_faulty_slbs = slb_ptr + (mmu_slb_size * i);
+		paca[i].aux_ptr->mce_faulty_slbs = slb_ptr + (mmu_slb_size * i);
 #endif
 }
 
